### ISSUE TITLE
systemd-networkd interfaces remain configuring when UseAddress=false 

### ISSUE THREAD

**Issue by seblu** [systemd/systemd#19196]
**systemd version the issue has been seen with**
 > systemd 248 (248-1-arch)
+PAM +AUDIT -SELINUX -APPARMOR -IMA +SMACK +SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN +IPTC +KMOD +LIBCRYPTSETUP +LIBFDISK +PCRE2 -PWQUALITY +P11KIT -QRENCODE +BZIP2 +LZ4 +XZ +ZLIB +ZSTD +XKBCOMMON +UTMP -SYSVINIT default-hierarchy=unified               

**Used distribution**
 > Arch Linux

**Linux kernel version used** (`uname -a`)

 > Linux hostname 5.11.0-seblu #1 SMP PREEMPT Tue Feb 16 15:56:00 CET 2021 x86_64 GNU/Linux

**CPU architecture issue was seen on**
 > x86_64

**Steps to reproduce the problem**
A very simple double interface situation where eth0 is providing the PD to a dummy interface.

#### eth0.network
```ini
[Match]
Name=eth0

[Network]
Address=1.2.3.4/24
Address=1::2/64
Gateway=1.2.3.1
DHCP=ipv6

[DHCPv4]
# Used by DHCPv6 but not allowed in DHCPv6
ClientIdentifier=duid-only
DUIDType=link-layer
DUIDRawData=xxx

[DHCPv6]
UseAddress=false
WithoutRA=solicit
```

#### dummy0.netdev
```ini
[NetDev]
Name=dummy0
Kind=dummy
```

#### dummy0.network
```ini
[Match]
Name=dummy0
[Network]
IPv6PrefixDelegation=dhcpv6
```

**Unexpected behaviour you saw**
Interface `eth0` remain configuring in networkctl 

```
# networkctl 
IDX LINK   TYPE     OPERATIONAL SETUP       
1 lo     loopback carrier     unmanaged
2 eth0   ether    routable    configuring
3 dummy0 ether    routable    configured 
```

Running in debug log level, shows `link_check_ready()` is unhappy because there is no dynamic assignment. Which is the purpose of UseAddress=false.
```
# SYSTEMD_LOG_LEVEL=debug /usr/lib/systemd/systemd-networkd
sd-device-monitor: No permission to stat PID1's netns, unable to determine if we are in host netns: Permission denied
Bus bus-api-network: changing state UNSET → OPENING
sd-bus: starting bus bus-api-network by connecting to /run/dbus/system_bus_socket...
Bus bus-api-network: changing state OPENING → AUTHENTICATING
Registering bus object implementation for path=/org/freedesktop/LogControl1 iface=org.freedesktop.LogControl1
timestamp of '/etc/systemd/network' changed
dummy0: loaded dummy
dummy0: Creating
...
eth0: Configuring route: dst: n/a, src: n/a, gw: fe80::x:x, prefsrc: n/a, scope: global, table: main(254), proto: ra, type: unicast, nexthop: 0
eth0: Setting NDisc routes.
eth0: Removing old NDisc information obtained from fe80::x:x.
eth0: link_check_ready(): DHCP4, DHCP6 or IPv4LL is enabled but no dynamic address is assigned yet.
eth0: NDisc routes set.
eth0: Removing old NDisc information obtained from fe80::x:x.
eth0: link_check_ready(): DHCP4, DHCP6 or IPv4LL is enabled but no dynamic address is assigned yet.
...
eth0: link_check_ready(): DHCP4, DHCP6 or IPv4LL is enabled but no dynamic address is assigned yet.
```

Despite these messages, IPv6 PD works.

**Expected behaviour you didn't see**
```
# networkctl 
IDX LINK   TYPE     OPERATIONAL SETUP       
1 lo     loopback carrier     unmanaged
2 eth0   ether    routable    configured
3 dummy0 ether    routable    configured 
```

**Comment by yuwata** [comment]
Ah, indeed. Thanks. I will try to fix this next week.

**Comment by seblu** [comment]
Great news ! 

As the [provider](https://documentation.online.net/en/dedicated-server/network/ipv6/prefix) where tests are made also provide an *IA_NA* address with the *IA_PD* prefix, I noticed there is the same issue with the "regular" DHCPv6 addresses. It's not only related to prefix delegation.

**A similar eth0 configuration, without prefix delegation on dummy interface:**
```ini
[Match]
Name=eth0

[Network]
Address=<static v4>/24
Address=<static v6 from the dhcp prefix>/128
Gateway=<static gw>
DHCP=ipv6

[DHCPv6]
UseAddress=false
WithoutRA=solicit
```

**Same output in networkctl:**
```
# networkctl 
IDX LINK  TYPE     OPERATIONAL SETUP      
  1 lo    loopback carrier     unmanaged
  2 eth0  ether    routable    configuring
```

**Same output in debug logs:**
```
# SYSTEMD_LOG_LEVEL=debug /usr/lib/systemd/systemd-networkd
...
eth0: New device has no master, continuing without
eth0: Flags change: +UP +LOWER_UP +RUNNING +MULTICAST +BROADCAST
eth0: Link 2 added
eth0: udev initialized link
eth0: State changed: pending -> initialized
eth0: Saved original MTU: 1500
...
eth0: Remembering foreign address: fe80::***/64 (valid forever, preferred forever)
eth0: link_check_ready(): link is in initialized state.
eth0: Gained IPv6LL
eth0: link_check_ready(): link is in initialized state.
eth0: Remembering foreign address: =<static v6>/128 (valid forever, preferred forever)
eth0: link_check_ready(): link is in initialized state.
..
eth0: Remembering foreign address: <static v4>/24 (valid forever, preferred forever)
eth0: link_check_ready(): link is in initialized state.
...
eth0: Setting nomaster done.
eth0: Setting address genmode done.
eth0: Removing route: dst: n/a, src: n/a, gw: fe80::**, prefsrc: n/a, scope: global, table: main(254), proto: ra, type: unicast, nexthop: 0
eth0: State changed: initialized -> configuring
eth0: Bringing link up
eth0: Discovering IPv6 routers
NDISC: Started IPv6 Router Solicitation client
DHCPv6 CLIENT: Started in Managed mode
eth0: Acquiring DHCPv6 lease
eth0: link_check_ready(): static addresses are not configured.
eth0: link_check_ready(): static addresses are not configured.
eth0: Updating address: <static v4>/24 (valid forever, preferred forever)
eth0: Updating address: <static v6>/128 (valid forever, preferred forever)
eth0: Setting addresses
DHCPv6 CLIENT: Sent SOLICIT
DHCPv6 CLIENT: Next retransmission in 1s
eth0: Forgetting route: dst: n/a, src: n/a, gw: fe80::**, prefsrc: n/a, scope: global, table: main(254), proto: ra, type: unicast, nexthop: 0
eth0: Remembering updated address: <static v4>/24 (valid forever, preferred forever)
eth0: link_check_ready(): static addresses are not configured.
eth0: Remembering updated address: <static v6>/128 (valid forever, preferred forever)
eth0: link_check_ready(): static addresses are not configured.
eth0: Addresses set
Setting '/proc/sys/net/ipv6/conf/eth0/proxy_ndp' to '0'
eth0: Configuring route: dst: n/a, src: n/a, gw: <gateway v4>, prefsrc: n/a, scope: global, table: main(254), proto: static, type: unicast, nexthop: 0
eth0: Setting routes with gateway
eth0: Routes with gateway set
eth0: link_check_ready(): DHCP4, DHCP6 or IPv4LL is enabled but no dynamic address is assigned yet.
DHCPv6 CLIENT: Recv ADVERTISE
DHCPv6 CLIENT: Sent REQUEST
DHCPv6 CLIENT: Next retransmission in 935ms
DHCPv6 CLIENT: T1 expires in 2h 58min 45s
DHCPv6 CLIENT: T2 expires in 1d 22h 50min 31s
eth0: link_check_ready(): DHCP4, DHCP6 or IPv4LL is enabled but no dynamic address is assigned yet.
DHCPv6 CLIENT: Recv REPLY
NDISC: Sent Router Solicitation, next solicitation in 3s
NDISC: Received Router Advertisement: flags none preference medium lifetime 1800 sec
NDISC: Invoking callback for 'router' event.
eth0: Configuring route: dst: n/a, src: n/a, gw: fe80::**, prefsrc: n/a, scope: global, table: main(254), proto: ra, type: unicast, nexthop: 0
eth0: Setting NDisc routes.
eth0: Received remembered route: dst: n/a, src: n/a, gw: fe80::**, prefsrc: n/a, scope: global, table: main(254), proto: ra, type: unicast, nexthop: 0
eth0: NDisc routes set.
eth0: Removing old NDisc information obtained from fe80::**.
eth0: link_check_ready(): DHCP4, DHCP6 or IPv4LL is enabled but no dynamic address is assigned yet.

```

The requirement here, is to get the lease for the *IA_NA* address, which tell the router to forward all packet to the *IA_NA* /64 prefix to our host. Then, we can define which interface address we want statically.

**Comment by yuwata** [comment]
Fix is waiting in #19241.
